From 08ed578ccca8c57875702d22ec59554f9224cef8 Mon Sep 17 00:00:00 2001
From: Derek Foreman <derekf@osg.samsung.com>
Date: Fri, 8 Jul 2016 12:50:57 -0500
Subject: [PATCH 07/94] compositor-drm: Stop sending uninit data to the kernel

Valgrind noticed that we send uninit data to drmModeAddFB2.  While
the kernel should never read this (because of the plane format),
it's probably still nicer to zero the data before we send it.

Signed-off-by: Derek Foreman <derekf@osg.samsung.com>
Reviewed-by: Yong Bakos <ybakos@humanoriented.com>
---
 src/compositor-drm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/compositor-drm.c b/src/compositor-drm.c
index 32b11cc..7335a1d 100644
--- a/src/compositor-drm.c
+++ b/src/compositor-drm.c
@@ -393,7 +393,7 @@ drm_fb_get_from_bo(struct gbm_bo *bo,
 {
 	struct drm_fb *fb = gbm_bo_get_user_data(bo);
 	uint32_t width, height;
-	uint32_t handles[4], pitches[4], offsets[4];
+	uint32_t handles[4] = { 0 }, pitches[4] = { 0 }, offsets[4] = { 0 };
 	int ret;
 
 	if (fb)
-- 
1.9.1

